core/ephemeral_key_provider_opkssh.go

```diff
@@ -0,0 +1,48 @@
+package core
+
+import (
+	"bytes"
+	"errors"
+	"os/exec"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// OpksshEphemeralKeyProvider is a prototype implementation of EphemeralKeyProvider
+// that interacts with the opkssh CLI tool to obtain ephemeral SSH keys.
+type OpksshEphemeralKeyProvider struct {
+	// Path to the opkssh CLI executable
+	OpksshPath string
+}
+
+// GetEphemeralKey runs the opkssh CLI to obtain the ephemeral private key.
+func (p *OpksshEphemeralKeyProvider) GetEphemeralKey() ([]byte, error) {
+	if p.OpksshPath == "" {
+		p.OpksshPath = "opkssh"
+	}
+
+	cmd := exec.Command(p.OpksshPath, "key", "export", "--private")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return nil, errors.New("failed to obtain ephemeral key from opkssh: " + err.Error())
```

Comment:
Nerdlog uses `github.com/juju/errors`, so we should keep using that, and here it shoudl be `errors.Annotatef(err, "obtaining ephemeral key from opkssh")`